lib/static/new-ui/features/suites/components/TreeActionsToolbar/index.tsx

```diff
@@ -0,0 +1,189 @@
+
+    const isEditScreensAvailable = useSelector((state: State) => state.app.availableFeatures)
+        .find(feature => feature.name === EditScreensFeature.name);
+
+    const isSelectedAll = useMemo(() => {
+        for (const suiteId of rootSuiteIds) {
+            if (suitesStateById[suiteId].checkStatus !== CHECKED) {
+                return false;
+            }
+        }
+
+        return true;
+    }, [suitesStateById, rootSuiteIds]);
+
+    const isSelectedAtLeastOne = useMemo(() => {
+        for (const suiteId of rootSuiteIds) {
+            if (suitesStateById[suiteId].shouldBeShown && (suitesStateById[suiteId].checkStatus === CHECKED || suitesStateById[suiteId].checkStatus === INDETERMINATE)) {
+                return true;
+            }
+        }
+
+        return false;
+    }, [suitesStateById, rootSuiteIds]);
+
+    const isStaticImageAccepterEnabled = useSelector(getIsStaticImageAccepterEnabled);
+    const isGuiMode = useSelector(getIsGui);
+    const visibleImages: ImageEntity[] = useSelector(getVisibleImages);
+    const selectedImages: ImageEntity[] = useSelector(getSelectedImages);
+    const activeImages = isSelectedAtLeastOne ? selectedImages : visibleImages;
+
+    const isAtLeastOneAcceptable = activeImages.some(image => isAcceptable(image));
+    const isAtLeastOneRevertable = activeImages.some(image => isScreenRevertable({image, gui: isGuiMode, isLastResult: true, isStaticImageAccepterEnabled}));
+    const isUndoButtonVisible = isAtLeastOneRevertable && !isAtLeastOneAcceptable;
+
+    const selectedTestsCount = useMemo(() => {
+        let count = 0;
+
+        for (const browser of Object.values(browsersStateById)) {
+            if (browser.checkStatus === CHECKED) {
+                count++;
+            }
+        }
+
+        return count;
+    }, [browsersStateById]);
+
+    const handleSelectAll = (): void => {
+        if (isSelectedAll) {
+            dispatch(deselectAll());
+        } else {
+            dispatch(selectAll());
+        }
+    };
+
+    const handleExpandAll = (): void => {
+        dispatch(expandAll());
+    };
+
+    const handleCollapseAll = (): void => {
+        dispatch(collapseAll());
+    };
+
+    const handleRun = (): void => {
+        if (isSelectedAtLeastOne) {
+            dispatch(retrySuite(selectedTests));
+        } else {
+            const visibleTests = visibleBrowserIds.map(browserId => ({
+                testName: browsersById[browserId].parentId,
+                browserName: browsersById[browserId].name
+            }));
+            dispatch(retrySuite(visibleTests));
+        }
+    };
+
+    const handleUndo = (): void => {
+        const acceptableImageIds = activeImages
+            .filter(image => isScreenRevertable({image, gui: isGuiMode, isLastResult: true, isStaticImageAccepterEnabled}))
+            .map(image => image.id);
+
+        if (isStaticImageAccepterEnabled && !isGuiMode) {
```

Comment:
If you were using `isStaticImageAccepterEnabled` as selector from `state.staticImageAccepter.enabled` instead of it being selector from `state.config`, you wouldn't need to be checking "if using gui" here (because `state.staticImageAccepter.enabled` is false in GUI)

Reply:
Removed those.